### DISCUSSION TITLE
Apps cant load data when using wireguard mode because client doesnt trust certificate

### DISCUSSION THREAD

**Discussion by matthewmturner** [mitmproxy/mitmproxy#5913] [Q&A] [answered]
So I used to be able to use wireguard mode without issue on my iPhone but i just recently got a new phone (iPhone 14 Pro Max) and now whenever i turn on wireguard all my apps cant connect to the internet because of certificate trust issues.  Not sure if it's actually related to getting a new phone or if thats coincidence.

For example I get the below - with the same error on all apps (the host is obviously different for each app).

```
Client TLS handshake failed. The client does not trust the proxy's certificate for edge-mqtt.facebook.com (sslv3 alert bad certificate)
```

Any pointers on how to fix this?

**Comment by mhils** [comment, marked as answer]
Did you install the mitmproxy CA certificate on your device, and did you follow _all_ instructions on mitm.it?

**Reply by pomozoff** [reply]
It seems a bug in iOS (16.3 at least)
Right after installing a mitmproxy profile I enabled the mitmproxy cert in Settings -> General -> About -> Certificate Trust Settings, however, mitmproxy did not pass traffic showing the error in the topic comment. Well, I disabled and enabled the cert  toggle again, and iOS warned me with the default alert about CA certs installation. Right after that mitmproxy started working as expected.

**Reply by matthewmturner** [reply]
@pomozoff indeed, it was related to uninstall and reinstall the cert a couple times to get it working. thanks!
